### PR TITLE
[22.05] xsd fixes for output actions

### PR DESCRIPTION
**Pull request by bernt-matthias**
Would like to backport since we are quite close to get `planemo` support 22.05 .. 

- `when` tags can contain `action` and `conditional` (i.e. `ActionsElement` and not only `Action`) https://github.com/galaxyproject/galaxy/blob/f1f16517e6384f680ebbcea0f0ffb2b871a3d36c/lib/galaxy/tool_util/parser/output_actions.py#L46
- `strip` is a string https://github.com/galaxyproject/galaxy/blob/f1f16517e6384f680ebbcea0f0ffb2b871a3d36c/lib/galaxy/tool_util/parser/output_actions.py#L481
- iterate attribute was missing

all used for instance here:

used for instance here: https://github.com/bgruening/galaxytools/blob/c27a05e92e6e03545903dd2ff90976f8cab5ebf7/tools/text_processing/text_processing/cut.xml#L69

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  - lint the cut tool
  - install planemo and replace the installed xsd file by this one
  - lint the cut tool

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]

**Comment by mvdbeek** [comment]
The doc build failure looks legit: https://github.com/galaxyproject/galaxy/actions/runs/4102363773/jobs/7075229114#step:12:28

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 1]
```diff
@@ -5645,22 +5645,30 @@ Source of the tabular data ``from_data_table``, ``from_param``, or ``from_file``
     </xs:attribute>
     <xs:attribute name="name" type="xs:string">
       <xs:annotation>
-        <xs:documentation xml:lang="en"></xs:documentation>
+        <xs:documentation xml:lang="en"><![CDATA[
+Name of the referred data table, parameter, of file (required).
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
Name of the referred data table, parameter, or file (required).
```

**Comment by bernt-matthias** [comment]
Looks good to me. Test failures appear to be unrelated?

File: lib/galaxy/tool_util/xsd/galaxy.xsd (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: Is that a rendering glitch, or did you revert the commit somehow ? Seems the typo is still there

Comment (bernt-matthias) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: My mistake. Thanks for double checking.

**Pull request merged**